[sig-storage] Run storage e2e test_client_pod as privileged

Elbehery opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind bug
/kind storage
/kind failing-test


<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

Fix running storage e2e tests in SELinux environment. 

HostPath Volume Plugin creates a directory within `/tmp` on host machine, to be mounted as volume.  Usually, `Inject-pod` writes content to the volume, and a `Client-pod` tried the read the contents and verify. 

When SELinux is enabled on the host, `Client-pod` can not read the content, with `permission denied`. 

This PR run the `Client-pod` as privileged, so that it can access the volume content, even when SELinux is enabled on the host. 

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes https://github.com/kubernetes/kubernetes/issues/84585

```release-note
Fixes hostpath storage e2e tests within SELinux enabled env
```

gnufied commented:
IIRC privileged containers don't get to bypass SELinux policies. So if SELinux was blocking them before, it will block them after. Please test carefully before going ahead with this approach.

Elbehery commented:
> IIRC privileged containers don't get to bypass SELinux policies. So if SELinux was blocking them before, it will block them after. Please test carefully before going ahead with this approach.

@gnufied  i have tested this on a cluster with SELinux enabled and the test passed successfully 
cc @vrutkovs @bertinatto

ping035627 commented:
/ok-to-test

bertinatto commented on test/e2e/framework/volume/fixtures.go:
```diff
@@ -511,8 +511,9 @@ func TestVolumeClientSlow(f *framework.Framework, config TestConfig, fsGroup *in
 }
 
 func testVolumeClient(f *framework.Framework, config TestConfig, fsGroup *int64, fsType string, tests []Test, slow bool) {
+	privileged := true
```
@Elbehery, thank you for working on this.  

I have a question for you. Although this fixes the issue,  could you check if you can set `privileged = true` only when the HostPath plugin is being used? Perhaps we could add a flag for that in the plugin definition.

The idea is to avoid always going privileged. Thanks!

Commit:
Run storage hostpath e2e test client pod as privileged

hostPath volume plugin creates a directory within /tmp on host machine, to be mounted as volume.
inject-pod writes content to the volume, and a client-pod tried the read the contents and verify.
when SELinux is enabled on the host, client-pod can not read the content, with permission denied.
running the client-pod as privileged, so that it can access the volume content, even when SEinux is enabled on the host.
